### PR TITLE
fix: Show result of waiting execution on canvas after execution complete

### PR DESCRIPTION
**Pull request by michael-radency**
## Summary

after manual execution finished execution result of waiting execution should be seen on canvas
image

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/NODE-1714/show-result-of-waiting-execution-on-canvas-after-execution-finished

### CODE REVIEW

File: packages/editor-ui/src/components/Node.vue [hunk 1]
```diff
@@ -320,9 +321,23 @@ const nodeTitle = computed(() => {
...
 		const lastNodeExecuted = get(workflowExecution, 'data.resultData.lastNodeExecuted');
 		if (props.name === lastNodeExecuted) {
+			const node = props.workflow.getNode(lastNodeExecuted);
+			if (
+				node &&
+				node.type === WAIT_NODE_TYPE &&
+				['webhook', 'form'].includes(node.parameters.resume as string)
+			) {
+				const eventType =
+					node.parameters.resume === 'webhook' ? 'incoming webhook call' : 'form submission';
```
Comment (elsmr) [inline packages/editor-ui/src/components/Node.vue:334]: these strings should also be in en.json

File: packages/editor-ui/src/components/Node.vue [hunk 2]
```diff
@@ -320,9 +321,23 @@ const nodeTitle = computed(() => {
 const waiting = computed(() => {
 	const workflowExecution = workflowsStore.getWorkflowExecution as ExecutionSummary;
 
-	if (workflowExecution?.waitTill) {
+	if (workflowExecution?.waitTill && !workflowExecution?.finished) {
```
Comment (elsmr) [inline packages/editor-ui/src/components/Node.vue:324]: These changes should probably also apply to the new canvas
`packages/editor-ui/src/composables/useCanvasMapping.ts` L317

Comment (michael-radency) [reply packages/editor-ui/src/components/Node.vue]: this condition is actually not needed, I did remove it

File: packages/editor-ui/src/composables/useRunWorkflow.ts [hunk 3]
```diff
@@ -263,42 +272,23 @@ export function useRunWorkflow(useRunWorkflowOpts: { router: ReturnType<typeof u
...
+					nodes: workflowData.nodes,
+					runData: workflowsStore.getWorkflowExecution?.data?.resultData?.runData,
+					destinationNode: options.destinationNode,
+					pinData,
+					directParentNodes,
+					source: options.source,
+					getTestUrl,
+				});
+			} catch (error) {
+				console.log(error);
```
Comment (ShireenMissi) [inline packages/editor-ui/src/composables/useRunWorkflow.ts:290]: was this intentional ?

File: packages/editor-ui/src/plugins/i18n/locales/en.json [hunk 4]
```diff
@@ -1011,6 +1014,8 @@
 	"node.nodeIsExecuting": "Node is executing",
 	"node.nodeIsWaitingTill": "Node is waiting until {date} {time}",
 	"node.theNodeIsWaitingIndefinitelyForAnIncomingWebhookCall": "The node is waiting for an incoming webhook call (indefinitely)",
+	"node.theNodeIsWaitingWebhookCall": "The node is waiting for an incoming webhook call",
+	"node.theNodeIsWaitingFormCall": "The node is waiting for an form submission",
```
Comment (ShireenMissi) [inline packages/editor-ui/src/plugins/i18n/locales/en.json:1018]:
```suggestion
	"node.theNodeIsWaitingFormCall": "The node is waiting for a form submission",
```

**Review by ShireenMissi — COMMENTED** [review]
Mostly look good, only one issue with new canvas while running the workflow you can see the line coming out of wait node green
Screenshot 2024-10-02 at 15 14 36
this works fine with the old canvas

File: packages/editor-ui/src/composables/useRunWorkflow.ts (hunk 3 above)
Comment (michael-radency) [reply packages/editor-ui/src/composables/useRunWorkflow.ts]: no, removed

**Review by ShireenMissi — APPROVED** [review]
LGTM 🎉

**Review by ShireenMissi — APPROVED** [review]

**Review by ShireenMissi — APPROVED** [review]

**Review by ShireenMissi — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.63.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.63.0)

**Comment by janober** [comment]
Got released with [`n8n@1.63.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.63.0)

**Pull request merged**